ext/gd/tests/gd_create_png_non_castable_user_stream.phpt

```diff
@@ -0,0 +1,85 @@
+--TEST--
+imagecreatefrompng(): uncastable user stream
+--EXTENSIONS--
+gd
+--XFAIL--
+Script hangs for unknown reason
+--SKIPIF--
+<?php
+if (!GD_BUNDLED) die("skip requires bundled GD library\n");
+?>
+--FILE--
+<?php
+
+class DummyStreamWrapper
+{
+    /** @var resource|null */
+    public $context;
+
+    /** @var resource|null */
+    public $handle;
+
+    public function stream_cast(int $castAs)
+    {
+        return false;
+    }
+
+    public function stream_close(): void { }
+
+    public function stream_open(string $path, string $mode, int $options = 0, ?string &$openedPath = null): bool
+    {
+        return true;
+    }
+
+    public function stream_read(int $count)
+    {
+        return 0;
```

Comment:
`::stream_read()` is supposed to return `string|false`. The `0` is converted to `"0"`, so we happily keep on reading:

https://github.com/php/php-src/blob/4cee2c0127431a7ebee2adc53be10b5b199f6344/main/streams/streams.c#L1522-L1531

Anyhow, `imagecreatefrompng()` is not supposed to reach the stream cast

https://github.com/php/php-src/blob/4cee2c0127431a7ebee2adc53be10b5b199f6344/ext/gd/gd.c#L1565

This only works for `imagecreatefromxbm()` and `imagecreatefromxpm()`. Using the former reports:
````
Warning: imagecreatefromxbm(): Cannot represent a stream of type user-space as a STDIO FILE* in %s on line %d
````
Maybe something is different on Windows.

Reply:
Indeed, same issue on Linux so changed to ``imagecreatefromxbm()``